ci-operator/step-registry/upi/aws/post/upi-aws-post-chain.yaml

```diff
@@ -0,0 +1,9 @@
+chain:
+  as: upi-aws-post
+  steps:
+  - chain: gather
+  - ref: upi-deprovision-aws
+  - ref: ipi-deprovision-deprovision
+  - ref: upi-deprovision-aws-check-destroy
```

Comment:
I expect reodrering these will remove the need to call `upi-deprovision-aws-check-destroy` and enable that functionality to be instead included in `upi-deprovision-aws`. By first calling `ipi-deprovision-deprovosion`, the cluster should mostly be cleaned up (and therefore unable to create new things). Then calling `upi-deprovision-aws` with the full funcionality in the `check-destroy` variant will finish things up. This will remove the need for `upi-deprovision-aws-check-destroy` altogether.

Reply:
I reordered it and now I get this cyclic dependency error here. 
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22317/rehearse-22317-pull-ci-openshift-installer-master-e2e-aws-upi/1485976419107344384

Reply:
I didn't look at this in too much detail; but it looks like possibly there is an object that's not getting tagged properly in the UPI template and therefore not deleted by the installer binary.